src/utils/swapErrorToUserReadableMessage.test.ts

```diff
@@ -0,0 +1,113 @@
+import { swapErrorToUserReadableMessage } from './swapErrorToUserReadableMessage'
+
+describe('swapErrorToUserReadableMessage', () => {
+  test('should return "Transaction rejected" when error code is 4001', () => {
```

Comment:
nit: AFAIK the rest of the codebase uses `it`, not `test`. I think it reads better ("it should...").